assignment.js

```diff
@@ -341,19 +337,21 @@ export function classifyNutrition() {
    */
   const extractDominantNutrition = (nutritionalData) => {
     const dominantNutrition = nutritionalData.reduce((acc, item) => {
-      Object.keys(item.nutritions).forEach((key) => {
+      Object.keys(item.nutritions).map((key) => {
         if (!acc[key] || acc[key].value < item.nutritions[key]) {
           acc[key] = {
             value: item.nutritions[key],
             type: item.type,
           };
         }
+        return undefined;
       });
       return acc;
     }, {});
     const categorizedNutrition = {};
-    Object.keys(dominantNutrition).forEach((key) => {
+    Object.keys(dominantNutrition).map((key) => {
       categorizedNutrition[key] = dominantNutrition[key].type;
+      return undefined;

```

Comment:
This is n ot a good usage of map here, looks like here we are modifying an object, should we not use reduce ?

Reply:
Yes Sir, making use of reduce than map was best in this situation.